ios/PolyPodApp/PodApi/PostOffice.swift

```diff
@@ -162,25 +173,39 @@ extension PostOffice {
     }
     
     private func handlePolyInDelete(args: [Any], completionHandler: @escaping (MessagePackValue?, MessagePackValue?) -> Void) {
-        let extendedDataSet = convertArgs(args: args, completionHandler: completionHandler)
+        let extendedDataSet: [ExtendedData]
+        do {
+            extendedDataSet = try convertArgs(args: args)
+        } catch {
+            completionHandler(nil, createErrorResponse(#function, error))
+            return
+        }
         
-        PodApi.shared.polyIn.deleteQuads(quads: extendedDataSet) { didDelete in
-            if didDelete {
-                completionHandler(MessagePackValue(), nil)
-            } else {
-                completionHandler(nil, MessagePackValue("Failed"))
+        PodApi.shared.polyIn.deleteQuads(quads: extendedDataSet) { error in
+            if let error = error {
+                completionHandler(nil, createErrorResponse(#function, error))
+                return
             }
+            completionHandler(MessagePackValue(), nil)
         }
     }
     
     private func handlePolyInHas(args: [Any], completionHandler: @escaping (MessagePackValue?, MessagePackValue?) -> Void) {
-        let extendedDataSet = convertArgs(args: args, completionHandler: completionHandler)
+        let extendedDataSet: [ExtendedData]
+        do {
+            extendedDataSet = try convertArgs(args: args)
+        } catch {
+            completionHandler(nil, createErrorResponse(#function, error))
+            return
+        }
         
         PodApi.shared.polyIn.hasQuads(quads: extendedDataSet) { doesHave in
             if doesHave {
                 completionHandler(MessagePackValue(), true)
             } else {
-                completionHandler(false, MessagePackValue("Failed"))
+                // TODO: This needs a closer look - should we really fail if the
+                //       data we check for is not there?
```

Comment:
This is _really_ suspicious -> If some data the feature wants to check for isn't there, we get an error... @shoniko - any idea why? I wouldn't want to change it with this PR, as the purpose of this one is only to improve error messages, but it seems like a bug to me.

Reply:
Yes, it does make sense to have something like this here indeed:

```
                completionHandler(MessagePackValue(), false)
```